### PR TITLE
KNI ready FakeBM infrastructure with scripts to deploy OpenShift 4.5 nightly build with baremetal IPI 

### PR DESCRIPTION
**Pull request by hgeaydem**
##### SUMMARY

This config creates a FakeBM environment for an Openshift Baremetal IPI and Openshift Virtualization lab

##### ISSUE TYPE

- New config Pull Request

##### COMPONENT NAME

kni-osp

##### ADDITIONAL INFORMATION

This config creates a FakeBM environment for Openshift deployment: 1 ospbmc host, 1 bastion host, 1 provision host, 3 masters and 3 workers and creates the install scripts on the provision node. It also provisions both a routed appnet network for floating IPs and an unrouted pxenet network for provisioning and allows to set port security on a per-node basis to allow for nested-virt networking to flow correctly without having to predict the mac addresses that will be used there 

```paste below

```

### CODE REVIEW

File: ansible/configs/kni-osp/default_vars_azure.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+################################################################################
+################################################################################
+### Azure Specific Variables
```
Comment (stencell) [inline ansible/configs/kni-osp/default_vars_azure.yml:4]: If this is an OSP specific config (`kni-osp`), is there a reason we have default vars for azure & ec2 as well?

File: ansible/configs/kni-osp/env_vars.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
...m1raUxDSnJkV0psY201bGRHVnpMbWx2TDNObGNuWnBZMlZoWTJOdmRXNTBMM05sWTNKbGRDNXVZVzFsSWpvaVpHVm1ZWFZzZEMxMGIydGxiaTAxZEdkbU55SXNJbXQxWW1WeWJtVjBaWE11YVc4dmMyVnlkbWxqWldGalkyOTFiblF2YzJWeWRtbGpaUzFoWTJOdmRXNTBMbTVoYldVaU9pSmtaV1poZFd4MElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WlhKMmFXTmxMV0ZqWTI5MWJuUXVkV2xrSWpvaVlqZzNNRGt4WmpZdE5qRXlNeTB4TVdVNUxXRTJNVGt0TkRJd01UQmhPR1V3TURBeUlpd2ljM1ZpSWpvaWMzbHpkR1Z0T25ObGNuWnBZMlZoWTJOdmRXNTBPbXR1YVRwa1pXWmhkV3gwSW4wLm51VGR0RlczRENHcFpvT0pCbU45VjQwWG1wbmlZRE9tUnI2Z05vNGVwRVBrb1lDXzk1YmhWX0ttYjhoTnprOTNVTGtDNnJXNTVjTXFQMVM4RHh3QWw0RUxRZ2NFZXIyalBJLXZBNGUzdlZ5cHNLbS1XSkFxcWo2OGhNN0Z4ekMzRGgxY19lN19EQkJLOWtxZmcyRzZiNTJXQmI2RUhsODg2Q2Nza3JBVm1fbmprNS14ay1Ma1hSM3lXNW5JeXlZdXhNVGg1LUNMd3lQQy1yLVIzeklzdnlWelNPVTgyeUJaaE1tUmc3enUtOWlydThENHdqRFJQclhiSm1FV3lBM1FIUlJ2VTJuci01MTFEeEhEbWhtNW14YU0tSFA4emk3SU8zVEU5SU55S3BqTmo5eTIwNmtFN0NNSVNMWmRWWFl3MkpIQ1BmSmJQMHNJY3V0dnFvOTdGdw==" }, "quay.io/openshift-cnv": {"username": "openshift-cnv+beta", "password": "HEEF8PBP6PKN33YINWP4ATD4P4VXKQFIWUKPU0XOCY22YI3LIHIA2H8YK0LQ0E7U", "auth": "b3BlbnNoaWZ0LWNuditiZXRhOkhFRUY4UEJQNlBLTjMzWUlOV1A0QVRENFA0VlhLUUZJV1VLUFUwWE9DWTIyWUkzTElISUEySDhZSzBMUTBFN1U="} } }'
```
Comment (stencell) [inline ansible/configs/kni-osp/env_vars.yml:61]: Is this a legit pull secret? This doesn't belong in a public repo.

File: ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2 [hunk 3]
```diff
@@ -0,0 +1,428 @@
+{
```
Comment (stencell) [inline ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2:1]: Is an azure cloud provider template actually needed for this config?

File: ansible/configs/kni-osp/files/cloud_providers/osp_cloud_template_master.j2 [hunk 4]
```diff
@@ -0,0 +1,262 @@
+heat_template_version: 2018-03-02
```
Comment (stencell) [inline ansible/configs/kni-osp/files/cloud_providers/osp_cloud_template_master.j2:1]: If you are using the OSP cloud provider and need to use your own HOT, please see the changes merged in https://github.com/redhat-cop/agnosticd/pull/1763/files.

We had to move some of the resource creation out of the HOT to make sure resources got created by the correct user (admin vs user).

File: ansible/configs/kni-osp/files/clouds.yaml.j2 [hunk 5]
```diff
@@ -0,0 +1,12 @@
+clouds:
+  {{ osp_project_name }}:
+    auth:
+      auth_url: "{{ osp_auth_url }}"
+      username: "{{ osp_auth_username }}"
```
Comment (stencell) [inline ansible/configs/kni-osp/files/clouds.yaml.j2:5]: I think this needs to be changed to `{{ guid }}-user`. While we use the full admin account to create some resources (i.e. project, user, etc), the project user should be used from that point forward and the admin credentials should never be written to the bastion - or at least not kept there if they must be used temporarily.

File: ansible/configs/kni-osp/files/clouds.yaml.j2 [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+      project_name: "{{ osp_project_name }}"
+      project_id: "{{ osp_project_id }}"
+      user_domain_name: "Default"
+      password: "{{ osp_auth_password }}"
```
Comment (stencell) [inline ansible/configs/kni-osp/files/clouds.yaml.j2:9]: same...this should be something like `{{ heat_user_password }}` and not the admin password.

File: ansible/configs/kni-osp/files/repos_template.j2 [hunk 7]
```diff
@@ -0,0 +1,44 @@
+[rhel-8-server-rpms]
```
Comment (stencell) [inline ansible/configs/kni-osp/files/repos_template.j2:1]: You might want to consider using the Satellite repo method. I don't believe we host any rhel 8 repos that can be accessed via the file method, but I am not 100% on this.

File: ansible/configs/ocp4-cluster/default_vars.yml [hunk 8]
```diff
@@ -28,11 +28,12 @@ install_ipa_client: false
 
 # Enable this if you want to create a user on the bastion
 # Mutually exclusive with {{ install_ipa_client }}
-install_student_user: true
+install_student_user: false
```
Comment (stencell) [inline ansible/configs/ocp4-cluster/default_vars.yml:31]: Please revert this change and do not include another config in this PR. If this needs to change, please open an issue or submit a separate PR so the config owner can address.

File: ansible/configs/ocp4-cluster/default_vars.yml [hunk 9]
```diff
@@ -28,11 +28,12 @@ install_ipa_client: false
...
 
 # This should be overwritten based on the user ordering the catalog item
 # It will be used by the bastion-student-user role and created on the bastion
 student_name: lab-user
+student_password: Redhat01
```
Comment (stencell) [inline ansible/configs/ocp4-cluster/default_vars.yml:36]: Please revert this change and do not include another config in this PR. If this needs to change, please open an issue or submit a separate PR so the config owner can address.

Also, we shouldn't be including passwords in defaults like this. I'd prefer this to be more like `CHANGEME`

File: ansible/configs/kni-osp/env_vars.yml (hunk 2 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/env_vars.yml:61]: I was told by gpte team members they would replace it with a generic one while getting it into a catalog item ? (when asking how I can get the user to provide his own)

File: ansible/configs/kni-osp/default_vars_azure.yml (hunk 1 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/default_vars_azure.yml:4]: no, removing those

File: ansible/configs/kni-osp/files/repos_template.j2 (hunk 7 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/repos_template.j2:1]: Can you please point me to that method ? I need several repos that I'm not even sure are available on sat ?

File: ansible/configs/ocp4-cluster/default_vars.yml (hunk 8 above)
Comment (hgeaydem) [reply ansible/configs/ocp4-cluster/default_vars.yml:31]: ocp4-cluster ain't mine/not using those in my config, these files have been pushed by mistake, sorry for that

File: ansible/configs/ocp4-cluster/default_vars.yml (hunk 9 above)
Comment (hgeaydem) [reply ansible/configs/ocp4-cluster/default_vars.yml:36]: ocp4-cluster ain't mine/not using those in my config, these files have been pushed by mistake, sorry for that

File: ansible/configs/kni-osp/files/clouds.yaml.j2 (hunk 6 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/clouds.yaml.j2:9]: this is how I found it in the tree I copied from, trying those changes

File: ansible/configs/kni-osp/files/clouds.yaml.j2 (hunk 5 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/clouds.yaml.j2:5]: this is how I found it in the tree I copied from, trying those changes

File: ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2 (hunk 3 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2:1]: no, removing those

File: ansible/configs/kni-osp/files/cloud_providers/osp_cloud_template_master.j2 (hunk 4 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/cloud_providers/osp_cloud_template_master.j2:1]: most of the changes there are meant to allow for some nodes having  2 interfaces and have th eability to turn posrt security off on some of those. Since I'm not creating any role/project or anything that should need admin permissions (which I don't have when testing in my OSP Sandbox environment anyway), this shouldn't be an issue, or is it ?

File: ansible/configs/kni-osp/files/clouds.yaml.j2 (hunk 6 above)
Comment (hgeaydem) [reply ansible/configs/kni-osp/files/clouds.yaml.j2]: leads to the following error when running the agnosticd config now : fatal: [provision]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'heat_user_password' is undefined"}

File: ansible/configs/kni-osp/env_vars.yml (hunk 2 above)
Comment (fridim) [reply ansible/configs/kni-osp/env_vars.yml]: Please make sure you contact the owner of that secret to revoke it ASAP

File: ansible/configs/kni-osp/default_vars_azure.yml (hunk 1 above)
Comment (NonyNo3lle) [reply ansible/configs/kni-osp/default_vars_azure.yml]: @stencell can you resolve if this is ok now?

File: ansible/configs/kni-osp/env_vars.yml (hunk 2 above)
Comment (NonyNo3lle) [reply ansible/configs/kni-osp/env_vars.yml]: @stencell can you resolve if this is ok now?

File: ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2 (hunk 3 above)
Comment (NonyNo3lle) [reply ansible/configs/kni-osp/files/cloud_providers/azure_cloud_template.j2]: @stencell can you resolve if this is ok now?

File: ansible/configs/kni-osp/files/repos_template.j2 (hunk 7 above)
Comment (NonyNo3lle) [reply ansible/configs/kni-osp/files/repos_template.j2:1]: set-repositories role takes repo_method: satellite 
https://github.com/redhat-cop/agnosticd/tree/development/ansible/roles/set-repositories
I am contacting ops to have them create the content view with the repositories listed in this j2 for you. See email sent.
